Source/MQTTnet/Internal/Disposable.cs

```diff
@@ -44,12 +44,13 @@ public void Dispose()
             {
                 return;
             }
+
+            _isDisposed = true;
```

Comment:
without this change the test ManagedClients_Will_Message_Send failed 

reason: the managed client sends a disconnect while it is disposed.

im not sure about this change. i